boards/DIM/Src/App/App_SevenSegDisplays.c

```diff
@@ -0,0 +1,51 @@
+#include <stdlib.h>
+
+#include "App_SevenSegDisplays.h"
+#include "App_SharedConstants.h"
+#include "App_SharedAssert.h"
+
+struct SevenSegDisplays
+{
+    float (*get_state_of_charge)(void);
+    void (*set_hex_digit[NUM_SEVEN_SEG_DISPLAYS_DIGITS])(uint8_t);
+};
+
+struct SevenSegDisplays *App_SevenSegDisplays_Create(
+    void (*const set_left_hex_digit)(uint8_t),
+    void (*const set_middle_hex_digit)(uint8_t),
+    void (*const set_right_hex_digit)(uint8_t))
+{
+    struct SevenSegDisplays *seven_seg_displays =
+        malloc(sizeof(struct SevenSegDisplays));
+
+    shared_assert(seven_seg_displays != NULL);
+
+    seven_seg_displays->set_hex_digit[SEVEN_SEG_DISPLAYS_LEFT_HEX_DIGIT] =
+        set_left_hex_digit;
+    seven_seg_displays->set_hex_digit[SEVEN_SEG_DISPLAYS_MIDDLE_HEX_DIGIT] =
+        set_middle_hex_digit;
+    seven_seg_displays->set_hex_digit[SEVEN_SEG_DISPLAYS_RIGHT_HEX_DIGIT] =
+        set_right_hex_digit;
+
+    return seven_seg_displays;
+}
+
+void App_SevenSegDisplays_Destroy(
+    struct SevenSegDisplays *const seven_seg_displays)
+{
+    free(seven_seg_displays);
+}
+
+void App_SevenSegDisplays_SetValue(
+    const struct SevenSegDisplays *const seven_seg_displays,
+    const uint8_t                        chars[],
+    size_t                               num_chars)
+{
+    shared_assert(num_chars <= NUM_SEVEN_SEG_DISPLAYS_DIGITS);
+
+    for (size_t i = 0; i < num_chars; i++)
+    {
+        shared_assert(chars[i] < NUM_HEX_DIGITS);
+        seven_seg_displays->set_hex_digit[i](chars[i]);
```

Comment:
What if I only give one character? Do the other ones remain unchanged?

Reply:
Oh yeah that's a good point. I am unsure what the best approach is. I can thinking we can either:
1. Leave the others unchanged
2. Default the unspecified digits to be zero
3. Default the unspecified digits to be turned off
4. The user has to specify exactly 3 chars each time

Which one do you think makes more sense?

Reply:
3

Reply:
I think if we want to go with 3, we will need another `HEX_DIGIT` for turning a display off (i.e., HEX_DIGIT_OFF). That way I can write
```
 ASSERT_EQ(
                    HEX_DIGIT_OFF,
                    set_left_hex_digit_fake.arg0_history[count - 1]);
```

But then that kinda breaks `enum HexDigit` because now we have 17 digits (`0-15`, `OFF`) to handle. What's an elegant way to get around this issue? Maybe give `Io_SevenSegDisplay` its own unique `enum SevenSegHexDigit` that has 17 digits rather than relying on `enum HexDigit` in `SharedConstant.h`?

Reply:
```
struct SevenSegHexDigit {
 bool enabled;
 HexDigit value;
}
```
maybe?

Reply:
Hmm i think that's about as elegant as we can get while having a generic hex enum. I'll try adding this in and make sure the comment is clear about what `enabled` means :v:

Reply:
Added the enabled bit. Essentially, if `enabled` is `false`, then `value ` is ignored entirely and we just turn off the display.